### DISCUSSION TITLE
How to generate a set of random locations across the map

### DISCUSSION THREAD

**Discussion by Dr-Goodbeard** [TUDelft-CNS-ATM/bluesky#465] [Q&A]
Hi, 

I have a plugin made that is randomly placing some number of aircraft across the map, Is there a way to do this same thing for locations? How can I completely clear the map of all the squares and triangles and then show only my set of randomly generate locations as triangles (or whatever shape)?

Thanks in advance,

**Comment by jooste** [comment]
Hi @nl22-nmsu, 

Indeed, traffic can be generated randomly using `MCRE`. There is no such built-in functionality for e.g., waypoints. It currently also isn't (easily) possible to disable loading of the default navigation database. If there is a good use case for it we can add it as a TODO.

You can, of course, already make a plugin to create random waypoints!

**Reply by Dr-Goodbeard** [reply]
Sure. I am able to generate the random waypoints using just lon and lat points, which can work for now. Eventually I want to be able to select the actual airports and only display those ports.

Where is all the airport data stored that gets loaded by the simulator?

And what command/function gets used to display the airport location (the little triangles). Can I use that to display just a subset of the airports that are there?

**Reply by jooste** [reply]
If you add waypoints with `DEFWPT` they will also become visible on the gui.

If you look in `bluesky/resources/navdata` you'll find all of the files that store the navigation data.

**Reply by Dr-Goodbeard** [reply]
Is it possible to show the waypoints/airports that I create and not all the other ones in the database? Or what functions could I use to make a plugin to do this?

**Reply by Dr-Goodbeard** [reply]
And during the Sims that I run it'd be nice to see where each aircraft was going. When I click on them I get the purple line that shows that. Is there a command to turn that feature on for all of the aircraft

**Reply by jooste** [reply]
It currently isn't possible to show only the custom waypoints, but you could implement this in a gui plugin. It is currently also only possible to show the route of one aircraft.

**Reply by Dr-Goodbeard** [reply]
What functions can I use for this? Is there a page on the wiki that discusses this?

**Reply by Dr-Goodbeard** [reply]
As a follow up/alternative, is it possible to change the color of the waypoints? The goal is just to make it easier to see what points on the map should be visited. Thanks.

**Reply by jooste** [reply]
This is unfortunately also not yet possible, but could be nice features!
